qiita_pet/handlers/analysis_handlers.py

```diff
@@ -321,3 +325,31 @@ def get(self):
                     user.shared_analyses + user.private_analyses]
 
         self.render("show_analyses.html", analyses=analyses)
+
+
+class ResultsHandler(StaticFileHandler, BaseHandler):
+    def validate_absolute_path(self, root, absolute_path):
+        """Overrides StaticFileHandler's method to include authentication
+        """
+        u = self.current_user
+
+        accessible_filepaths = get_accessible_filepath_ids(u)
+        accessible_filepaths = {join(get_db_files_base_dir(),
+                                     filepath_id_to_rel_path(fpid))
+                                for fpid in accessible_filepaths}
+
+        for accessible_filepath in accessible_filepaths:
+            access = False
+
+            if (accessible_filepath == absolute_path):
+                access = True
+
+            elif isdir(accessible_filepath) and \
+                    absolute_path.startswith(accessible_filepath + sep):
```

Comment:
I think it depends on the job's parameters (if the job specifies an output directory with a trailing slash, then maybe that's what gets recorded in the filepath table). In any case, this check is necessary because job results can be directories, and the static handler has to be able to serve the files from within those directories if they have access, but you wouldn't want to allow access to `5_summarize_taxa_through_plots.py_output_dir_buahahahaha` if the user only has access to `5_summarize_taxa_through_plots.py_output_dir`. I couldn't think of a better way to do this, but I'm open to suggestions!